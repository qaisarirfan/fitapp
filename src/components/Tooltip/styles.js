import {StyleSheet} from "react-native"

// Tooltip: Styling here
const styles = StyleSheet.create({
  container: {
    backgroundColor: "rgba(0, 0, 0, 0.9)",
    position: "absolute",
    bottom: "120%",
    left: 0,
    right: 0,
    marginHorizontal: 12,
    borderRadius: 5,
  },
  message: {
    padding: 12,
    color: "#fff",
  },
  arrow: {
    width: 0,
    height: 0,
    backgroundColor: "transparent",
    borderStyle: "solid",
    borderLeftWidth: 10,
    borderRightWidth: 10,
    borderBottomWidth: 10,
    borderLeftColor: "transparent",
    borderRightColor: "transparent",
    borderBottomColor: "rgba(0, 0, 0, 0.9)",
    transform: [{rotate: "180deg"}],
    position: "absolute",
    top: "100%",
    left: 12,
  },
})

export default styles
