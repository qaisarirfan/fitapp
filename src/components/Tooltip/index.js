import React from "react"
import {View, Text} from "react-native"
import PropTypes from "prop-types"
import styles from "./styles"

// Tooltip Component content
export const Tooltip = ({message}) => (
  <View style={styles.container}>
    <Text style={styles.message}>{message}</Text>
    <View style={styles.arrow} />
  </View>
)

// Tooltip Proptypes
Tooltip.propTypes = {
  message: PropTypes.string,
}

// Tooltip Default props
Tooltip.defaultProps = {
  message: "",
}

export default Tooltip
