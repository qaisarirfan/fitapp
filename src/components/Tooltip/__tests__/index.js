import React from "react"
import renderer from "react-test-renderer"
import Tooltip from ".."

const createProps = () => ({
  classes: {},
})

describe("<Tooltip />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<Tooltip {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
