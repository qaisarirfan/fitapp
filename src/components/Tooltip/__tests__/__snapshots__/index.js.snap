// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> Does not explode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FF8585",
      "bottom": "110%",
      "left": 0,
      "marginHorizontal": 12,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <Text
    style={
      Object {
        "padding": 12,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#FF8585",
        "borderBottomWidth": 15,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 15,
        "borderRightColor": "transparent",
        "borderRightWidth": 15,
        "borderStyle": "solid",
        "height": 0,
        "left": 12,
        "position": "absolute",
        "top": "100%",
        "transform": Array [
          Object {
            "rotate": "180deg",
          },
        ],
        "width": 0,
      }
    }
  />
</View>
`;
